packages/react-expandable-table/src/index.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright (c) Quid, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @flow
+import * as React from 'react';
+import styled from '@emotion/styled/macro';
+import { Icon } from '@quid/react-core';
+import { themes } from '@quid/theme';
+import { Tooltip } from '@quid/react-tooltip';
+import useControlledState from '@quid/react-use-controlled-state';
+import {
+  ColumnCell,
+  Row,
+  ExpandableContent,
+  List,
+  Header,
+  AngleButton,
+  Ellipsis,
+  InfoIcon,
+  SortIcon,
+  TooltipContainer,
+} from './styles';
+import type { Data, ID, SortOrder } from './types';
+import {
+  getSortedData,
+  onKeyboardSelect,
+  filterDataForPagination,
+} from './utils';
+
+type Column = {
+  label: string,
+  key: string,
+  align?: 'left' | 'right' | 'center',
+  width?: number,
+  tooltip?: React.Node,
+  bold?: boolean,
+};
+
+type RenderRow = ({
+  isOdd: boolean,
+  isLast: boolean,
+  data: Data,
+}) => React.Node;
+
+type Props = {
+  /** Set this if you'd like the table to handle the expanded/collapsed state internally */
+  defaultOpenedRows?: Array<ID>,
+  /** Set this if you'd like to manually handle the expanded/collapsed state */
+  openedRows?: Array<ID>,
+  /** You will need to manually handle the openedRows list using the list returned by this callback */
+  onToggle?: (Array<ID>) => void,
+  /**  By default only one row at a time can be expanded */
+  maxOpen?: number,
+  /** Define the columns structure, it will be used to order the data */
+  columns: Array<Column>,
+  /** The data to display in the table, refer to the `Data` type or the example for more information */
+  data: Array<Data>,
+  /**
+   * @deprecated Maximum list height, please prefer a custom `className` or `style` value instead
+   */
+  maxBodyHeight?: number,
+  /** Set this to a number higher than 0 to enable pagination */
+  page?: number,
+  /** When pagination is enabled, this is the maximum number of items visible on each page */
+  maxItemPerPage?: number,
+  /** Sorting order, you can invert this to have a different behavior */
+  defaultSortOrder?: SortOrder,
+  // FIXME: this logic is weak, we should have a better way to control the sorting
+  // maybe just like we do for the openedRows logic, where we allow both a controlled
+  // and uncontrolled way to use the functionality. Keeping as is for now for backward compat
+  /**
+   * @deprecated When set to `true`, the first column will be automatically sorted by
+   * the first order value defined in `defaultSortOrder`, it will be replaced by a
+   * more flexible implementation in the future
+   */
+  defaultIsSorting?: boolean,
```

Comment:
@gabrielmicko I'm not sure this is doing what the original component did, may you please review your implementation and describe what's the exact expected behavior? I couldn't figure it out from it.

If this is something we don't use I'd say we can directly remove it.

Reply:
It referred to if you want to sort on initial render or not. What we might want here is to say that which columns should be used for sorting on first render. That was missing from the other implementation as well.
```
  const [sorting, setSorting] = React.useState({
    key: defaultIsSorting ? columns[0].key : null, //-> maybe we should change defaultIsSorting to defaultSortBy so user can define which column should be sorted. What do you think?
    sort: defaultIsSorting ? defaultSortOrder[0] : null,
  });
```

Reply:
maybe we should change defaultIsSorting to defaultSortBy so user can define which column should be sorted. What do you think?

Reply:
I would rather remove this feature for now if we aren't using it, but I see we set it to `true` in our internal code so I wonder why...

Reply:
Remove please